src/state/service.rs

```diff
@@ -49,7 +50,13 @@ impl Service {
 #[derive(Debug, Eq, PartialEq, Clone, serde::Serialize, serde::Deserialize)]
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 pub struct Endpoint {
+    /// The service VIP for this endpoint.
+    pub vip: NetworkAddress,
```

Comment:
Added the vip to this structure to support the case of removing and then re-adding services. The VIP is needed to repopulate the waypoints to VIP map.

Reply:
I like this change; I much prefer immutable services to the on-the-heap initial implementation I did. thank you